tests/test_broadcast.py

```diff
@@ -248,6 +249,31 @@ async def test_broadcast_filter() -> None:
     assert (await receiver.receive()) == 15
 
 
+async def test_broadcast_filter_type_guard() -> None:
+    """Ensure filter type guard works."""
+    chan = Broadcast[int | str](name="input-chan")
+    sender = chan.new_sender()
+
+    def _is_int(num: int | str) -> TypeGuard[int]:
+        return isinstance(num, int)
+
+    # filter out all numbers less than 10.
+    receiver = chan.new_receiver().filter(_is_int)
```

Comment:
the comment needs to be updated, but looks great otherwise!

Reply:
Done!